### ISSUE TITLE
Room name doesn't register special characters instead of throwing error

### ISSUE THREAD

**Issue by puneetlath** [Expensify/App#12678]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Break down in numbered steps
1. Choose `New Room` from the create menu
2. Type in a special character (e.g. `!` or `.` in the room title

## Expected Result:
Describe what you think should've happened
The special character that was typed is shown, along with a validation error saying that character is not allowed. 

## Actual Result:
Describe what actually happened
The character is not shown at all. 

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** v1.2.26-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Issue reported by:** @puneetlath 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1668173276175579?thread_ts=1668112427.854869&cid=C049HHMV9SM

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by puneetlath** [comment]
@lschurr I'll fix this.

**Comment by puneetlath** [comment]
Will start on this today.

**Comment by lschurr** [comment]
Cool @puneetlath - do I need to do anything? Should I be trying to reproduce?

**Comment by puneetlath** [comment]
Nope, it's reproducible.

**Comment by lschurr** [comment]
Heyooooo any updates on this one @puneetlath?

**Comment by puneetlath** [comment]
I realized I need to refactor this page to use the new Forms component, so it's a little bit more involved then I initially thought. Hope to have a PR up today.

**Comment by puneetlath** [comment]
Still working on this, should be done soon.

**Comment by lschurr** [comment]
@puneetlath - you're OOO this week right?

**Comment by lschurr** [comment]
Gonna move to Weekly since Puneet is OOO.

**Comment by lschurr** [comment]
Hey hey @puneetlath - any update on this one?

**Comment by puneetlath** [comment]
Working on it today! ETA Monday.

**Comment by puneetlath** [comment]
Still in progress. Working through an issue [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1670710080656329).

**Comment by muttmuure** [comment]
Also assigning myself since I'm managing the tracking issue [here](https://github.com/Expensify/App/issues/13623)

**Comment by puneetlath** [comment]
Still on hold, now for https://github.com/Expensify/App/issues/13915

**Comment by puneetlath** [comment]
@davidcardoza we need some error copy for when someone tries to enter an invalid room name. We only allow letters, numbers, and dashes in room names. All other characters are disallowed. I'm thinking: 

>Room names cannot contain punctuation or most special characters

<img width="362" alt="image" src="https://user-images.githubusercontent.com/1371996/210663821-25f8f40a-0abc-407e-8be5-40a03b49d19e.png">


For reference, here is Slack's: 

<img width="500" alt="image" src="https://user-images.githubusercontent.com/1371996/210663902-c6ebec26-9518-4062-abad-a793fe29b993.png">

How does that sound?

**Comment by davidcardoza** [comment]
How about this, just a slight addition:
>Room names cannot contain spaces, punctuation or most special characters

I think most users will try to create a room with spaces, but that's just my opinion.

**Comment by puneetlath** [comment]
If they try to enter a space we automatically replace it with a dash. So I don't think we need to include that.

**Comment by davidcardoza** [comment]
That's awesome, I didn’t know that. Your copy looks good to me in that case.

**Comment by puneetlath** [comment]
@lschurr FYI this is internal, but @rushatgabhane is helping out with PR review and will need to be paid out (once this goes live).

**Comment by lschurr** [comment]
@rushatgabhane could you apply for the job in Upwork? https://www.upwork.com/jobs/~01f8eb2fdbe370c1ac

**Comment by lschurr** [comment]
Hey @puneetlath - Is this live? Should I pay and close this out or are we still waiting on something?

**Comment by puneetlath** [comment]
Yes! The [PR hit production](https://github.com/Expensify/App/pull/13994#issuecomment-1376037972) on Jan 9 so you're good to pay out @rushatgabhane on this one.

**Comment by lschurr** [comment]
Awesome. Paid!
